src/main/java/au/org/ala/biocache/service/AlaLayersService.java

```diff
@@ -239,7 +245,83 @@ public String getLayersServiceUrl() {
 
     @Override
     public Reader sample(String[] analysisLayers, double[][] points, Object o) {
-        // TODO: for on the fly intersection of layers not indexed
+        int TIMEOUT = 300000; // the layer service should complete intersection in 5 minutes
+        String fields =StringUtils.join(analysisLayers, ",");
+        String flatPoints = Arrays.toString(Arrays.stream(points)
+                .flatMapToDouble(Arrays:: stream)
+                .toArray());
+        String p = flatPoints.substring(1,flatPoints.length() -1); //remove []
+        String requestBody = "fids=" + fields + "&points=" + p;
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.TEXT_PLAIN);
+        HttpEntity<String> request = new HttpEntity<String>(
+                requestBody, headers);
+
+        try {
+            ResponseEntity<String> taskResponse = restTemplate.postForEntity(this.layersServiceUrl + "/intersect/batch", request, String.class);
+
+            if (taskResponse.getStatusCode() == HttpStatus.OK) {
+
+                JSONObject taskResult = JSONObject.fromObject(taskResponse.getBody());
+                String statusUrl = taskResult.getString("statusUrl");
+
+                if (!Strings.isNullOrEmpty(statusUrl)) {
+                    logger.debug("Checking intersect status: " + statusUrl);
+                    final long deadline = System.currentTimeMillis() + TIMEOUT;
+                    boolean done = false;
+                    do {
+                        ResponseEntity<String> statusResponse = restTemplate.getForEntity(statusUrl, String.class);
+                        if (statusResponse.getStatusCode() == HttpStatus.OK) {
+                            JSONObject processResult = JSONObject.fromObject(statusResponse.getBody());
+
+                            String status = processResult.getString("status");
+                            if (status.equalsIgnoreCase("finished")) {
+                                String downloadUrl = processResult.getString("downloadUrl");
+                                logger.debug("Reading intersect result: " + downloadUrl);
+                                URL url = new URL(downloadUrl);
+                                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```

Comment:
Connection not closed anywhere.

Reply:
Yes, that one is tricky. The solution I can figure out is to store zip to local temp folder and close the connection, and then go to next step